components/scream/src/share/io/scorpio_output.hpp

```diff
@@ -0,0 +1,485 @@
+#ifndef SCREAM_SCORPIO_OUTPUT_HPP
+#define SCREAM_SCORPIO_OUTPUT_HPP
+
+#include <iostream>
+#include <fstream>
+
+#include "scream_config.h"
+#include "ekat/ekat_parameter_list.hpp"
+#include "ekat/std_meta/ekat_std_enable_shared_from_this.hpp"
+#include "ekat/mpi/ekat_comm.hpp"
+#include "ekat/std_meta/ekat_std_utils.hpp"
+#include "ekat/ekat_parse_yaml_file.hpp"
+
+#include "share/io/scream_scorpio_interface.hpp"
+
+#include "share/field/field_repository.hpp"
+#include "share/field/field_header.hpp"
+#include "share/field/field.hpp"
+#include "share/field/field_identifier.hpp"
+
+#include "share/grid/grids_manager.hpp"
+/* There are a number of things that need to be cleaned up - in no particular order -
+ *--1) Gather dimension data from grid manager or field repo.  We may need to use field repo since grid manager
+ *     is unaware of say the number of components, for example.  --DONE--
+ *--2) Clean-up init step, perhaps with subfunctions.  --DONE--
+ *--3) Gather degree's of freedom from grid manager. Need to know the global-id index for
+ *     each value in a view, in order if flattened.  --DONE--
+ *  4) Handle MPI communication to allow for PIO_STRIDE (i.e. fewer pio ranks than total ranks).
+ *  5) Make it so that all dimensions that are registered are automatically registered as variables and the values written (only once, not with time).  Probably need this information from the grid manager(?)
+ *  6) Write Restart output.  Extra-savy would be to use DAG from AD to determine which fields were essential and just write those.  #3
+ * 6a) Also need restart history, in case you stop mid-out frequency and need to pick it up in the next run.
+ *--7) Create an AtmosphereInput class built on these same concepts.  Can we make a master SCORPIO class with output and input as sub-classes?  --DONE--
+ *--8) Create average, min, max and instantaneous options for output.  --DONE--
+ *  9) Assure that IO is compatible with PACKing of variables.
+ * 10) Add control of netCDF header information to the YAML file and this class.
+ * 11) Better naming convention for output files?  Currently just keeps a counter, but in EAM the date/time range is used.
+ * 12) Better handling of output frequency.  Need to a) gather time info from driver and b) parse that info for frequency units.
+ * 13) Expand compatability of IO class to handle Dynamics grids.  This will require a different approach to "set_dofs"
+ *-14) Hook up the AD for the unit test.  --DONE--
+ * 15) SCORPIO in stand-alone build.
+ * 16) When dynamics is ready, write output from a dynamics run.
+ * 17) Remove the F90 layer and call PIO C routines directly.
+ * 18) Add option to restart to write a restart at finalization, maybe if OUT_N = -1
+ * 19) take advantage of shared pointers field_repo and gridsmanager to streamline init, run and finalize interface.
+ */
+
+namespace scream
+{
+
+class AtmosphereOutput 
```

Comment:
Again, a brief description of the type and its purpose, and comments for its methods would be nice, and probably easier to do sooner rather than later.